crates/router/src/connector/adyen.rs

```diff
@@ -86,6 +86,13 @@ impl ConnectorValidation for Adyen {
             ),
         }
     }
+    fn validate_psync_reference_id(
+        &self,
+        _data: &types::PaymentsSyncRouterData,
+    ) -> CustomResult<(), errors::ConnectorError> {
+        // since we can make psync call with our reference_id, having connector_transaction_id is not an mandatory criteria
+        Ok(())
```

Comment:
Adyen supports PSync only incase of redirect flow. So we should validate the encoded data in the request, if it is not present we should not allow PSync call to the connector